_posts/2019-01-08-Golang.md

```diff
@@ -0,0 +1,168 @@
+many programming languages.
+
+TVM import and compilation process generates a graph json, a module and a params. Any application that
+integrate TVM runtime can load these compiled module and infer. Detailed tutorial of module import and
+compilationcan using TVM can be found at [tutorials](https://docs.tvm.ai/tutorials/).
+
+TVM now supports deploying compiled modules through Golang. Golang applications can make use of this 
+to deploy the deep learning models through TVM. Scope of this blog is introduce ```gotvm``` package,
+building the package, sample application build on ```gotvm``` to load a compiled module and infer.
+
+## Package
+
+Golang package ```gotvm``` is build on top of TVM's C runtime interface. The API in this package
+abstracts the native C types and provides Golang compatible types. Package source can be found
+at [gotvm](https://github.com/dmlc/tvm/tree/master/golang).
+
+This package leverages golangs interface, slices, function closures and impliceltly handles the
+necessary conversions across API calls.
+
+{:center: style="text-align: center"}
+![image](/images/golang/TVM-Golang-Blog.jpg){: width="60%"}
+{:center}
+<center> Golang Interface over TVM Runtime </center> <p></p>
+
+## How to
+
+As shown in below diagram ```gotvm``` enables golang applications integrate deep learning models
+from various frameworks with out the hassle of understanding each framework related interface API.
+Developers can make use of TVM to import and compile deep learning models and generate TVM artifacts.
+```gotvm``` package provides golang friendly API to load, configure, feed input and get output.
+
+{:center: style="text-align: center"}
+![image](/images/golang/TVM-Golang-Flow.jpg){: width="80%"}
+{:center}
+<center> Import, Compile, Integrate and Deploy</center> <p></p>
+
+TVM [Compile Deep Learning Models](https://docs.tvm.ai/tutorials/#compile-deep-learning-models) tutorial
+are available to compile models from all frameworks supported by TVM frontend. This compilation process
+generates the artifacts required to integrate and deploy the model on target.
+
+## API
+
+```gotvm``` package provides handfull of Datatypes and API to initialize, load and infer
+from a golang application. Like any other golang package we just need to import ```gotvm``` package here.
+
+- Module : Module API can be used to load a TVM compiled module into TVM runtime and access any functions.
+- Value : Value API provides helper function to set arguments or get return values in golang types like basic types or slices.
+- Function : Function API is useful to get handles for various function handles and invoke the same.
+- Array : Array API is useful to set or get Tensor data via golang slice.
+- Context : Context contain helper functions to build backend context handles.
+
+## Example
+
+Below shown is a simple example with inline documentation to load a compiled module and infer the same.
```

Comment:
A simple example with inline documentation of loading a compiled module and performing inference is shown below.